pystac_client/client.py

```diff
@@ -195,21 +201,78 @@ def from_file(  # type: ignore
             )
 
         client: Client = super().from_file(href, stac_io)
-
-        client._stac_io._conformance = client.extra_fields.get(  # type: ignore
-            "conformsTo", []
-        )
         client.modifier = modifier
 
         return client
 
-    def _supports_collections(self) -> bool:
-        return self._conforms_to(ConformanceClasses.COLLECTIONS) or self._conforms_to(
-            ConformanceClasses.FEATURES
+    def has_conforms_to(self) -> bool:
+        return bool(self._stac_io and "conformsTo" in self.extra_fields)
+
+    def get_conforms_to(self) -> List[str]:
+        return cast(List[str], self.extra_fields.get("conformsTo", []).copy())
+
+    def set_conforms_to(self, conformance_classes: List[str]) -> None:
+        self.extra_fields["conformsTo"] = conformance_classes
+
+    def add_conforms_to(
+        self, conformance_class: Union[ConformanceClasses, str]
+    ) -> None:
```

Comment:
```suggestion
    def add_conforms_to(
        self, *conformance_classes: str
    ) -> None:
```

Can we use arbitrary argument lists to remove our `isinstance` checks?

Reply:
I think the class is poorly named. It is just an Enum class where each name corresponds to the value of a conformance url. The `str` just makes things a little easier on the user (and the cli) so we can do either of these:

```python
client.add_conforms_to(ConformanceClasses.FILTER)
client.add_conforms_to("FILTER")
```